- 結果画面のグラフをレーダーチャートに変更

YanaiShohei opened the pull request:
- 問題のジャンル別の正解数をカウントができるように変更
- 結果画面のグラフをレーダーチャートに変更

YanaiShohei committed:
- 結果画面のグラフをレーダーチャートに変更
- 問題のジャンル別の正解数をカウントができるように変更

hikaru-nakamura-thedott commented:
コンフリクトが起きてるので解決してくださーい。

hikaru-nakamura-thedott commented:
@YanaiShohei 
ジャンルごとに正解数をカウントするオブジェクトが宣言されていますが、これらの変数名やジャンル名はまだ決めないんですか？
https://github.com/YanaiShohei/intern-yanai/pull/62/files#diff-43e48298f803a0385d05f1340fc02101e754484b84f22f3ceafaecfbd5d9c9a4R16-R23

hikaru-nakamura-thedott commented:
result画面に遷移した時に点数が表示されてないので直してくださーい。
コンソールに出てるエラーも見てねー。

YanaiShohei committed:
Merge branch 'develop' into feature/#39

YanaiShohei committed:
- quizCountの位置が間違えていたので修正しました。

YanaiShohei committed:
Merge branch 'feature/#39' of github.com:YanaiShohei/intern-yanai into feature/#39

YanaiShohei commented:
@hikaru-nakamura-thedott 
ジャンルの変数名は、まだ出題する問題のジャンルが決めきれていないので、後のissueの時に変更しようと思っています。

YanaiShohei committed:
quizCountの挙動がおかしくなってしまったので再調整しました。

YanaiShohei commented:
result画面の点数表示で修正しようとしたのですが、上のrouterLink？から飛んでしまうと表示されないかもしれないです。
title画面から開始ボタンを押す→question画面で問題を解き終える。→result画面
の手順なら点数が表示されました。

hikaru-nakamura-thedott commented:
@YanaiShohei 
> ジャンルの変数名は、まだ出題する問題のジャンルが決めきれていないので、後のissueの時に変更しようと思っています。

承知しました。
issueの内容に対して、実際の更新内容が中途半端に飛躍してるので、この辺はごちゃごちゃしないように早めに決めた方がいいですね。

中途半端な修正を行うと以下のようなデメリットもあるので、issueごとにキリのいいところまでやりましょう。
- コミット内容と実際の修正・更新に乖離が生まれやすくなるので後々変更が見づらくなる
- 「どこまでやったか」「次はどこからやるのか」がわからなくなりやすい。
- 上記からチグハグなソースになってバグの原因とかになりやすい
- いらないコードが残ってしまう

> result画面の点数表示で修正しようとしたのですが、上のrouterLink？から飛んでしまうと表示されないかもしれないです。
title画面から開始ボタンを押す→question画面で問題を解き終える。→result画面
の手順なら点数が表示されました。

挙動確認しました。確かにquestion -> resultなら正常に動きました。ありがとうございます！

hikaru-nakamura-thedott commented:
@YanaiShohei 
また何点か確認・指摘です。

1. ジャンルごとのカウント、問題を一周したらリセットしないんですか？
    - そこも後でやるんでしょうか？
2. おそらく以前にデバッグした時にあちこちでconsole.log出力してますが、もう必要のなくなったものは削除してください。
    - 本当に必要なconsole.logが要らないのに紛れて見つけづらくなります。
    - 今のうちに綺麗にしましょう。

hikaru-nakamura-thedott commented on app/src/app/question/question.component.ts:
```diff
@@ -1,6 +1,6 @@
 import { Component, OnInit } from '@angular/core';
 import { UrlHandlingStrategy } from '@angular/router';
-import { Quiz, Choice, QUIZ_DATA} from 'src/app/const/quiz';
+import { Quiz, Choice, QUIZ_DATA, } from 'src/app/const/quiz';
```
ここのカンマ必要？

hikaru-nakamura-thedott commented on app/src/app/services/quiz.service.ts:
```diff
@@ -1,5 +1,5 @@
 import { Injectable } from '@angular/core';
-import { Router } from '@angular/router';
+import { ChildrenOutletContexts, Router } from '@angular/router';
```
ChildrenOutletContexts 使ってる？
使ってないなら削除しましょう。

hikaru-nakamura-thedott commented on app/src/app/services/quiz.service.ts:
```diff
@@ -30,20 +38,46 @@ export class QuizService {
     return this.quizCount;
   }
 
+  // 正解した問題をカウントする
```
ナイスコメント 👍

hikaru-nakamura-thedott commented on app/src/app/const/quiz.ts:
```diff
@@ -14,6 +14,8 @@ export interface Quiz {
   //explanationというstring型の変数 ヒントを格納?
   hint: string;
+  // 問題のジャンル
+  category: string;
```
ジャンルなの？カテゴリなの？

hikaru-nakamura-thedott replied:
これはカテゴリーになりそうだね。

hikaru-nakamura-thedott commented on app/src/app/question/question.component.ts:
```diff
@@ -41,6 +41,7 @@ export class QuestionComponent implements OnInit {
   }
 
   // quizServiceから関数を呼び出す
+
```
ここはコメントも書いてあるし、空白行はない方がいいねー

hikaru-nakamura-thedott reviewed (changes requested):
細かいところ見つけてしまいました。

YanaiShohei committed:
-不要になったコンソールログ出力を削除
-問題を一周したらカテゴリー別の正解数カウントを０にする。
-ChildrenOutletContexts と UrlHandlingStrategyが使われていなかったので削除
とを削除
-行やカンマの修正

YanaiShohei committed:
- カテゴリーの名前を変更
- 結果画面のグラフに問題画面で解いた問題の結果を反映させる
- グラフのサイズを調整

hikaru-nakamura-thedott commented on app/src/app/result/result.component.html:
```diff
@@ -1,6 +1,6 @@
 <h1>点数</h1>
 <h2>{{this.trueAnswerCount}}/{{this.quizList.length}}</h2>
-<div width="400" height="400">
+<div style="height:400px; width:400px">
```
おお！すばらしい！
この書き方 `<div width="400" height="400">` はhtml5では非推奨の書き方になっているので、styleプロパティを使う形にしてくれて嬉しいです！

hikaru-nakamura-thedott commented on app/src/app/services/quiz.service.ts:
```diff
@@ -61,24 +61,24 @@ export class QuizService {
-      if(this.quizList[this.quizCount].category == 'カテゴリー5'){
-        ++this.correctAnswerCountCategory.Category5
+      if(this.quizList[this.quizCount].category == 'gen2020'){
+        ++this.correctCategoryCount.gen2020
```
連想配列を使っているので、以下のように書き換えても同じ結果になります！
`correctCategoryCount.gen2020`  -> `correctCategoryCount['gen2020']`
これを利用してこのif文の連続をもっと短く書いてみましょう！

hikaru-nakamura-thedott replied:
通話で話したようなことをやろうとすると、
>++this.correctCategoryCount[this.quizList[this.quizCount].category]

「this.correctCategoryCountのkeyの方が何型か指定されていないからダメ！」みたいな感じで怒られる。
<img width="1680" alt="スクリーンショット 2022-01-12 19 45 27" src="https://user-images.githubusercontent.com/86763776/149126414-3fc18756-a31d-4704-944a-571ffb8d6b2c.png">


this.correctCategoryCountのkeyの型を定義してあげましょう。
ついでにnumber型の値を持つことも定義してあげます。

<img width="463" alt="スクリーンショット 2022-01-12 19 51 52" src="https://user-images.githubusercontent.com/86763776/149126952-ab38b0bc-873d-4e6a-87b3-28efbd52b150.png">

このようにすることで `++this.correctCategoryCount[this.quizList[this.quizCount].category]` でアクセスできるようになります。

YanaiShohei committed:
カテゴリー別の正解数カウントのif文の連続を短く
